index.js

```diff
@@ -45,6 +45,17 @@ function createWriteHead (prevWriteHead, listener) {
   }
 }
 
+/**
+ * Emit 'headers' event on response object.
+ */
+
+function emitHeadersEvent() {
+  if (!this.headersEventEmitted) {
+    this.headersEventEmitted = true
+    this.emit('headers')
```

Comment:
This could be a `Symbol` to avoid conflicts with any other library that might be emitting a `headers` event.